types/AbstractBaseType.php

```diff
@@ -385,6 +386,33 @@ public function compare($column, $comp, $value) {
     }
 
     /**
+     * This function is used to modify an aggregation query to add a filter
+     * for the given column matching the given value. A type should add at
+     * least a filter here but could do additional things like joining more
+     * tables needed to handle more complex filters
+     *
+     * @param QueryBuilder $QB the query so far
+     * @param string $column The column name to use in the SQL
+     * @param string $comp The comparator @see Search::$COMPARATORS
+     * @param string $value this is the user supplied value to compare against
+     * @param string $type the logical operator this filter should use (AND|OR)
+     */
+    public function filter(QueryBuilder $QB, $column, $comp, $value, $type) {
+        $pl = $QB->addValue($value);
+
+        switch($comp) {
+            case '~':
+                $QB->filters()->where($type, "$column LIKE $pl");
+                break;
+            case '!~':
+                $QB->filters()->where($type, "$column NOT LIKE $pl");
+                break;
+            default:
+                $QB->filters()->where($type, "$column $comp $pl");
+        }
+    }
```

Comment:
In the current master  the ``$column`` argument for ``compare()`` in ``Search::getSQL()`` is already constructed to consist of the right aliases/column references. This should be done for the ``filter()`` method as well. However I wonder more about the ``$value`` argument. What if that actually refers to a value in column of another table as well instead of a given fix value?

I think it would be helpful for me, if we can write down some concrete use cases which need to be covered, rather than talk about it in abstract terms. I see if I can get started with this today.

Some style questions:
- Could we change the ``$type`` parameter to ``$op`` or ``$operator``? It gets confusing because we already are in class named ``Type``.
- I'm not sure about this myself: But would it be a good idea to put the line ``$QB->filters()->where($type, "$column $comp $pl");`` after the switch and have the switch only contain lines like ``$comp = 'LIKE';``?